validator/src/main/java/com/google/daq/mqtt/util/ConfigManager.java

```diff
@@ -275,4 +279,26 @@ public String getUpdatedTimestamp() {
     return isoConvert(metadata.timestamp);
   }
 
+  private DiscoveryConfig getDiscoveryConfig() {
+    DiscoveryConfig discoveryConfig = new DiscoveryConfig();
+    discoveryConfig.families = new HashMap<>();
+
+    if (metadata.discovery != null && metadata.discovery.families != null) {
+      metadata.discovery.families.keySet().forEach(family -> {
+        FamilyDiscoveryConfig familyDiscoveryConfig = new FamilyDiscoveryConfig();
+        FamilyDiscoveryModel familyModel = metadata.discovery.families.get(family);
+
+        if (familyModel != null) {
+          familyDiscoveryConfig.generation = familyModel.generation;
```

Comment:
One thing registrar does is some light error checking on things in the model, just to provide clues as to what might be wrong as early in the process as possible. Towards that, take a look at the discovery docs: https://github.com/faucetsdn/udmi/blob/master/docs/specs/sequences/discovery.md#sporadic-scan

If you chart it out, it's somewhat of an error in this case if a generation is supplied for a sporadic scan (meaning scan_interval_sec is not defined) in the sense of this being a model. So, maybe add some logic here to detect that case and flag an error. This will also get you slowly introduced to some of the logical behavior of the system. Something like:

`checkState(familyModel.generation != null && familyModel.scan_interval_sec == null, "generation specified without scan_interval_sec parameter")`